legend-engine-authentication-pure/src/main/resources/core_authentication/pure/vX_X_X/models/metamodel.pure

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import meta::protocols::pure::vX_X_X::metamodel::authentication::*;
+
+Class <<typemodifiers.abstract>> meta::protocols::pure::vX_X_X::metamodel::authentication::AuthenticationSpec
+{
+    _type    : String[1];
+}
+
+Class meta::protocols::pure::vX_X_X::metamodel::authentication::OauthAuthenticationSpec extends meta::protocols::pure::vX_X_X::metamodel::authentication::AuthenticationSpec
+{
+    grantType: String[1];
+    clientId: String[1];
+    clientSecretVaultReference: String[0..1];
+    authServerUrl: String[1];
+}
+
+Class meta::protocols::pure::vX_X_X::metamodel::authentication::VaultCredential extends meta::protocols::pure::vX_X_X::metamodel::authentication::AuthenticationSpec
```

Comment:
Is this model defined correctly?

Reply:
Fixed. To unload the credential concept, i introduced something called as Credential Provider(this is abstract). 

Class meta::pure::runtime::connection::authentication::CredentialVault extends meta::pure::runtime::connection::authentication::CredentialProvider
{
    vaultReference: String[1];
}